frontend/src/components/pagedItemsTable/PagedItemsTable.tsx

```diff
@@ -117,23 +118,109 @@ function PagedItemsTable<T extends object>({
 		[dispatch, setPageSizeCallback]
 	)
 
+	const shiftHeld = useRef(false)
+	useEffect(() => {
+		const handleKeyDown = (event: KeyboardEvent) => {
+			if (event.key === 'Shift') {
+				shiftHeld.current = true
+			}
+		}
+		const handleKeyUp = (event: KeyboardEvent) => {
+			if (event.key === 'Shift') {
+				shiftHeld.current = false
+			}
+		}
+		document.addEventListener('keydown', handleKeyDown)
+		document.addEventListener('keyup', handleKeyUp)
+		return () => {
+			document.removeEventListener('keydown', handleKeyDown)
+			document.removeEventListener('keyup', handleKeyUp)
+		}
+	}, [])
+
 	// We use this callback to respond when the user sorts a column
-	const sortByCallback: TableProps<any>['onChange'] = useCallback(
-		(pagination, filters, sorterResult) => {
-			if (!Array.isArray(sorterResult)) {
-				const sorter = sorterResult
-				const key = sorter.columnKey?.toString()
-				const order = sorter.order ?? undefined
-				if (key) {
-					if (sortBy === undefined || key !== sortBy.key || order !== sortBy.order) {
-						setSortByCallback({ key, order })
-					}
+	const sortByCallback = useCallback((columnKey: string) => {
+		const [existingFirst = undefined, existingSecond = undefined] = pagedItems.sortByList
+		const newFirst: Partial<SortBy> = { ...existingFirst }
+		const newSecond: Partial<SortBy> = { ...existingSecond }
+	
+		if (!shiftHeld.current) {
+			if (existingFirst?.key === columnKey) {
+				if (existingFirst.order !== 'descend') {
+					newFirst.order = 'descend'
+				} else {
```

Comment:
Lot's of ifs ... what about always setting the key and using an enum with a numeric order on a modulo 3 ?